tests/test_regex.py

```diff
@@ -74,5 +300,37 @@ def test_named_groups(self):
     def test_begining(self):
         assert_can_generate('^abc')
 
+    def test_caret_in_the_middle_does_not_generate_anything(self):
+        r = re.compile('a^b')
+
+        with pytest.raises(he.NoSuchExample):
+            h.find(regex(r), r.match)
```

Comment:
You probably out of context here. Thing is that this regex "a^b" should not match anything. You can try that. But the way generator works currently it would generate strings "ab", which is incorrect. So I added this test and update generation to prevent generation in that case.